src/components/profile/Profile.jsx

```diff
@@ -13,6 +14,26 @@ export default class Profile extends React.Component {
     };
 
     this.handleUpdating = this.handleUpdating.bind(this);
+    this.getIdAirTable = this.getIdAirTable.bind(this);
+  }
+
+  DestroyUser () {
+    this.getIdAirTable(this.props.student.id);
+    base('Profil').destroy(this.state.idAirTable, function (err, deletedRecord) {
```

Comment:
Ici ça ne va pas.
`this.getIdAirtable()` est asynchrone, donc la mise à jour du `state` ne va pas être effective qd vous appelez `destroy()`.
Il faut appeler `destroy()` dans un callback que vous passez à `getIdAirTable()`, même pas besoin de passer par le `state`